pkg/reconciler/build/build.go

```diff
@@ -44,14 +53,16 @@ type Reconciler struct {
 	// buildclientset is a clientset for our own API group
 	buildclientset clientset.Interface
 
-	buildsLister listers.BuildLister
+	buildsLister                listers.BuildLister
+	buildTemplatesLister        listers.BuildTemplateLister
+	clusterBuildTemplatesLister listers.ClusterBuildTemplateLister
 
 	// Sugared logger is easier to use but is not as performant as the
 	// raw logger. In performance critical paths, call logger.Desugar()
 	// and use the returned raw logger instead. In addition to the
 	// performance benefits, raw logger also preserves type-safety at
 	// the expense of slightly greater verbosity.
-	Logger *zap.SugaredLogger
+	logger *zap.SugaredLogger
```

Comment:
You could consider including Base Reconciler inside build reconciler like [this](https://github.com/knative/serving/blob/master/pkg/reconciler/v1alpha1/service/service.go#L44-L51). You get declaration cruft for free

Reply:
Unfortunately we define that base in `knative/serving` :-/